lib/api/apiUtils/object/versioning.js

```diff
@@ -133,6 +133,11 @@ function processVersioningState(reqVersionId, mst, vstat, cb) {
             options.versionId = '';
             options.isNull = true;
             options.dataToDelete = mst.objLocation;
+            // if null version exists, clean it up prior to put
+            if (mst.isNull) {
+                delOptions.versionId = mst.versionId;
+                return cb(null, options, null, delOptions);
```

Comment:
It's true that this is an issue of atomicity.
We can solve it using the existing BATCH operation of metadata, but we need update this operation to support versioning; we did not do it before because S3 does not use BATCH.